src/platform/plugins/shared/workflows_management/public/features/workflow_context/lib/get_variables_schema.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { getStepId } from '@kbn/workflows';
+import type { WorkflowGraph } from '@kbn/workflows/graph';
+import { DataSetStepTypeId } from '@kbn/workflows-extensions/common';
+import { z } from '@kbn/zod/v4';
+
+function inferZodTypeFromValue(value: unknown): z.ZodTypeAny {
```

Comment:
Wouldn't it be possible to reuse `inferZodType`?

https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/workflows_management/common/lib/zod/infer_zod_type.ts